engine/src/main/java/org/hibernate/search/query/engine/impl/HSQueryImpl.java

```diff
@@ -238,7 +238,7 @@ private TimeoutManagerImpl getTimeoutManagerImpl() {
 			if ( luceneQuery == null ) {
 				throw new AssertionFailure( "Requesting TimeoutManager before setting luceneQuery()" );
 			}
-			timeoutManager = new TimeoutManagerImpl( luceneQuery, timeoutExceptionFactory, this.extendedIntegrator.getTimingSource() );
+			timeoutManager = new TimeoutManagerImpl( String.valueOf( luceneQuery ), timeoutExceptionFactory, this.extendedIntegrator.getTimingSource() );
```

Comment:
BTW the reasoning is that the string is only needed when and ever if the timeout exception is thrown.